### PR TITLE
PMA - Filters TM-4138

### PR DESCRIPTION
**Pull request by scott062**

### CODE REVIEW

**Comment by scott062** [comment]
Still waiting to figure out how the org filter will work - will probably require mock and api PRs

File: src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx [hunk 1]
```diff
@@ -112,7 +119,7 @@ const PanelMeetingAgendas = (props) => {
   const { data: categories, loading: categoryLoading } = useDataLoader(api().get, '/fsbid/panel/reference/categories/');
   const categoriesOptions = uniqBy(sortBy(get(categories, 'data.results'), [(c) => c.mic_desc_text]), 'mic_desc_text');
   // Replace with real ref data endpoints
-  const { data: orgs, loading: orgsLoading } = useDataLoader(api().get, '/fsbid/agenda_employees/reference/handshake-organizations/');
+  const { data: orgs, loading: orgsLoading } = useDataLoader(api().get, '/fsbid/agenda_employees/reference/current-organizations/');
```
Comment (elizabeth-jimenez) [inline src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx:122]: 🦜

File: src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx [hunk 2]
```diff
@@ -152,30 +159,39 @@ const PanelMeetingAgendas = (props) => {
...
-      { 'legs.languages.code': code }
-    ));
-    const grades$ = selectedGrades.map(({ code }) => (
-      { 'legs.grade': code }
-    ));
+    const languages$ = selectedLanguages.flatMap(({ code }) => ([
+      { 'legs.languages.code': code },
+      { 'user.languages.code': code },
+    ]));
+    const grades$ = selectedGrades.flatMap(({ code }) => ([
```
Comment (elizabeth-jimenez) [inline src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx:166]: ;)

File: src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx [hunk 3]
```diff
@@ -152,30 +159,39 @@ const PanelMeetingAgendas = (props) => {
...
     if (textSearch) {
       const t = textSearch;
       // See Fuse extended search docs
       const freeTextLookups = [
-        { id: `'${t}` },
-        { 'assignment.pos_num': `'${t}` },
+        { 'assignment.pos_num': `^${t}` },
         { 'assignment.pos_title': t },
-        { 'legs.pos_num': `'${t}` },
+        { 'legs.pos_num': `^${t}` },
```
Comment (elizabeth-jimenez) [inline src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx:194]: what was the motivation behind this?

Comment (scott062) [reply src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx:194]: Before I was doing includes - ex.
Target: #123456789
Match: #6789
-
Now its start with - ex.
Target: #123456789
Match: #123
- 
Basically nobody should be typing in only the last part of the num - they should start with the identical beginning match and if they haven't finished typing then it'll still match and give them hints at where to keep going

**Comment by elizabeth-jimenez** [comment]
I'm wondering if we should remove the Search button? Since the searching is Live? Not done checking, so this could be for an edge-case i'm just not seeing yet

File: src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx (hunk 3 above)
Comment (elizabeth-jimenez) [reply src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx:194]: "nobody should be typing in only the last part of the num" that makes complete sense!

**Comment by elizabeth-jimenez** [comment]
Seeing some weird behavior with the Remarks dropdown - when i go down the list and select for the other dropdowns, it is smooth, but not for Remarks. Hard to explain, so lemme know if you see it. Just go down and start selecting them one by one.

File: src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx [hunk 4]
```diff
@@ -354,7 +374,7 @@ const PanelMeetingAgendas = (props) => {
               ref={childRef}
               textSearch={textSearch}
               label="Find Agenda Item"
-              placeHolder="Search using Panel Meeting Agenda Item Info"
+              placeHolder="Search for Agenda using Position, Name, or Official Item #"
```
Comment (elizabeth-jimenez) [inline src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx:377]: ❤️

**Comment by elizabeth-jimenez** [comment]
image
should we standardize by adding Item to Category and removing it from Action?

File: src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx [hunk 5]
```diff
@@ -185,6 +201,10 @@ const PanelMeetingAgendas = (props) => {
         { 'updaters.first_name': t },
         { 'updaters.emp_user.emp_user_last_name': t },
```
[the remark this answers is not in the document]
Comment (scott062) [reply src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx:202]: Non emp vs emp I believe unless our mapping is consistent

**Comment by scott062** [comment]
> I'm wondering if we should remove the Search button? Since the searching is Live? Not done checking, so this could be for an edge-case i'm just not seeing yet

We shouldn't be using it probably - should I remove it?

**Comment by elizabeth-jimenez** [comment]
> > I'm wondering if we should remove the Search button? Since the searching is Live? Not done checking, so this could be for an edge-case i'm just not seeing yet
> 
> We shouldn't be using it probably - should I remove it?

Yea! 💪

**Comment by elizabeth-jimenez** [comment]
> > image should we standardize by adding Item to Category and removing it from Action?
> 
> Or do we want to remove `Item` from everywhere?

That works!

File: src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (elizabeth-jimenez) [reply src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx:202]: ahhhh gotcha!

**Comment by elizabeth-jimenez** [comment]
Free Text: 
- [x] Position Num look up starts with (and works for positions that start with a 0)
- [x] Position name
- [x] Creator
- [x] Updater
- [x] Bidder
- [x] CDO
- [x] Official item num

- [x] Bureau (not seeing all in dropdown, but the ones that are do filter properly) (current assignment)
- [x] Org (only for position)
- [x] Category
- [x] Grade - filters on person grade
- [x] Grade - filters on position grade
- [x] Action
- [x] Status
- [x] Language - filters on person
- [x] Language - filters on position
- [ ] Remarks (seeing some weird behavior with the dropdown, but the filtering is g2g)
- [x] Skill - just person


- [x] Or within filters And between them

**Review by elizabeth-jimenez — CHANGES_REQUESTED** [review]
Just a few small changes

**Comment by elizabeth-jimenez** [comment]
I really like this fuse search - it is SO fast; great job Scott!

File: src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx [hunk 6]
```diff
@@ -466,7 +488,7 @@ const PanelMeetingAgendas = (props) => {
                   value={selectedRemarks}
                   options={remarksOptions}
                   onChange={setSelectedRemarks}
-                  valueKey="rc_code"
+                  valueKey="seq_num"
```
Comment (scott062) [inline src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx:491]: Fixes dropdown bug

File: src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx [hunk 7]
```diff
@@ -354,7 +374,8 @@ const PanelMeetingAgendas = (props) => {
               ref={childRef}
               textSearch={textSearch}
               label="Find Agenda Item"
-              placeHolder="Search using Panel Meeting Agenda Item Info"
+              placeHolder="Search for Agenda using Position, Name, or Official Item #"
+              noButton
```
[the remark this answers is not in the document]
Comment (elizabeth-jimenez) [reply src/Components/Panel/PanelMeetingAgendas/PanelMeetingAgendas.jsx:378]: nah, this is fine!

**Review by elizabeth-jimenez — APPROVED** [review]
🐈‍⬛

**Pull request merged**